### PR TITLE
mocking tests

### PR DESCRIPTION
**Pull request by hfurkanbozkurt**

### CODE REVIEW

File: doc/source/gyms.rst [hunk 1]
```diff
@@ -5,10 +5,77 @@ Algorithms
...
+
+The state of the environment is described by followings:
+
+- Normalized_delta:
+	Average difference between mean of f score values in last step and lowest number of unsatisfied clauses so far.
+
+- Normalized_ham:
+	Average Hamming distance between last configuration and current configuration.
+
+- Best_dalte:
```
Comment (AndreBiedenkapp) [inline doc/source/gyms.rst]: I think this should be delta instead of dalte

File: doc/source/gyms.rst [hunk 2]
```diff
@@ -5,10 +5,77 @@ Algorithms
...
+- Conflicts_analyzed: 
+	Number of conflicts the solver has analyzed in last step.
+
+- Restarts: 
+	Restart policy of the solver with parametrization.
+
+- Restarts_last: 
+	Number of restarts solver performed in last step.
+
+In each step, restarts parameter is controlled.
```
Comment (AndreBiedenkapp) [inline doc/source/gyms.rst]: By default

File: doc/source/gyms.rst [hunk 3]
```diff
@@ -5,10 +5,77 @@ Algorithms
...
+Although restarts parameter is descriebd as <policy><parametrization>, the action is applied directly to parametrization without changing the policy. When step function is used with a dict, both policy and parametrization can be controlled.
+
+Refer to `Potassco guide  <https://github.com/potassco/guide/releases/>`_ for more information.
 
 SAPS
 ------
-What parameters are controllable?
-What instances are available?
\ No newline at end of file
+
```
Comment (AndreBiedenkapp) [inline doc/source/gyms.rst]: We should state that we follow the Battiti paper

File: test/general_clingo_test.py [hunk 4]
```diff
@@ -1,16 +1,142 @@
 import unittest
+from unittest.mock import patch
 
 import os
 import numpy as np
-import gym
+import gym, pcg
```
Comment (AndreBiedenkapp) [inline test/general_clingo_test.py]: pcg is not needed here

File: test/general_clingo_test.py [hunk 5]
```diff
@@ -1,16 +1,142 @@
...
 import mtrand
-import subprocess
 import random
+import glob
 
-class TestGym(unittest.TestCase):
+from pcg.envs import ClingoEnv
+
+class MockClingo(ClingoEnv):
+    def __init__(self, **kwargs):
```
Comment (AndreBiedenkapp) [inline test/general_clingo_test.py]: Might be possible to get a simpler Mock object

File: test/general_saps_test.py [hunk 6]
```diff
@@ -1,12 +1,33 @@
...
+import clingo
 import mtrand
+import random
+import glob
 
-class TestGym(unittest.TestCase):
+from pcg.envs import SapsEnv
+
+class MockSAPS(SapsEnv):
+    def __init__(self, **kwargs):
```
Comment (AndreBiedenkapp) [inline test/general_saps_test.py]: MockClingo should probably be as simple as this Mock object

File: test/general_saps_test.py [hunk 7]
```diff
@@ -1,12 +1,33 @@
...
+        super().__init__(**kwargs)
```
Comment (AndreBiedenkapp) [inline test/general_saps_test.py]: Not sure if we really need to invoke the original saps

File: .gitattribute [hunk 8]
```diff
@@ -1,2 +1,2 @@
-pcg/envs/saps_core/src/* linguist-vendored
-pcg/envs/clingo_core/output* linguist-vendored
\ No newline at end of file
+* linguist-vendored
```
Comment (AndreBiedenkapp) [inline .gitattribute]: I'm not sure if this will work any better

File: doc/source/gyms.rst (hunk 3 above)
Comment (hfurkanbozkurt) [reply doc/source/gyms.rst]: Very interestingly, I fail to find the paper. Can you help me out?

Comment (hfurkanbozkurt) [reply doc/source/gyms.rst]: @AndreBiedenkapp (not sure if you receive a notification)

File: test/general_clingo_test.py (hunk 5 above)
Comment (hfurkanbozkurt) [reply test/general_clingo_test.py]: Any ideas on how simply we should be?

File: test/general_saps_test.py (hunk 6 above)
Comment (hfurkanbozkurt) [reply test/general_saps_test.py]: Same goes here...

File: test/general_clingo_test.py (hunk 5 above)
Comment (AndreBiedenkapp) [reply test/general_clingo_test.py]: If we can avoid real algorithm calls it would be great.
So probably just mocking the step function would suffice

**Pull request merged**